aten/src/ATen/native/cuda/foreach/Utils.cuh

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#define ILP 4
+#define CHUNK_SIZE 65536
+#define BLOCK_SIZE 1024
+namespace {
+
+// TensorListMetadata has to be < 4KB - the limit for kernel launch argument
+constexpr int depth_to_max_tensors[5] = {110, 64, 48, 36, 30};
+constexpr int depth_to_max_blocks[5] = {320, 320, 320, 320, 320};
+
+template<int n> struct TensorListMetadata
+{
+  void* addresses[n][depth_to_max_tensors[n-1]];
+  int sizes[depth_to_max_tensors[n-1]];
+  unsigned char block_to_tensor[depth_to_max_blocks[n-1]];
+  int block_to_chunk[depth_to_max_blocks[n-1]];
+};
+
+template<typename T>
+__device__ __forceinline__ bool is_aligned(T* p){
+  return ((uint64_t)p) % (ILP * sizeof(T)) == 0;
+}
+
+template<typename T>
+__device__ __forceinline__ void load_store(T* dst, T* src, int dst_offset, int src_offset){
+  typedef typename std::aligned_storage<ILP * sizeof(T), ILP * alignof(T)>::type LT;
```

Comment:
This seems to be at the heart of the Windows failure. cc @mcarilli @ngimel  - do you know of common mitigations / has this come up before within PyTorch? I couldn't find any old issues that seemed relevant.

The error message is

"""
error: static assertion failed with "You've instantiated std::aligned_storage<Len, Align> with an extended alignment (in other words, Align > alignof(max_align_t)). Before VS 2017 15.8, the member "type" would non-conformingly have an alignment of only alignof(max_align_t). VS 2017 15.8 was fixed to handle this correctly, but the fix inherently changes layout and breaks binary compatibility (*only* for uses of aligned_storage with extended alignments). Please define either (1) _ENABLE_EXTENDED_ALIGNED_STORAGE to acknowledge that you understand this message and that you actually want a type with an extended alignment, or (2) _DISABLE_EXTENDED_ALIGNED_STORAGE to silence this message and get the old non-conforming behavior."
"""

I'm not sure we want to enabled these macros and I also couldn't find mentions of them in the codebase so far.

Reply:
GPU pointwise kernels get away with just alignas: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/MemoryAccess.cuh#L128-L132, @mcarilli can you get away with the same thing, or you truly need std::aligned_storage?

Reply:
I don't know, @FDecaYed added the vectorized loads/stores.  I'll ping him.

Reply:
I think alignas should be just fine. std::aligned_storage happens to be the way I know so I picked it at the time. Looking at doc alignas might actual be the prefered way now.